modules/azure/servicebus.go

```diff
@@ -0,0 +1,383 @@
+package azure
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2017-04-01/servicebus"
+	"github.com/stretchr/testify/require"
+)
+
+func serviceBusNamespaceClientE(subscriptionID string) (*servicebus.NamespacesClient, error) {
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	nsClient := servicebus.NewNamespacesClient(subscriptionID)
+	nsClient.Authorizer = *authorizer
+	return &nsClient, nil
+}
+
+func serviceBusTopicClientE(subscriptionID string) (*servicebus.TopicsClient, error) {
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	tClient := servicebus.NewTopicsClient(subscriptionID)
+	tClient.Authorizer = *authorizer
+	return &tClient, nil
+}
+
+func serviceBusSubscriptionsClientE(subscriptionID string) (*servicebus.SubscriptionsClient, error) {
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	sClient := servicebus.NewSubscriptionsClient(subscriptionID)
+	sClient.Authorizer = *authorizer
+	return &sClient, nil
+}
+
+// ListServiceBusNamespaceE list all SB namespaces in all resource groups in the given subscription ID. This function would fail the test if there is an error.
+func ListServiceBusNamespaceE(subscriptionID string) (*[]servicebus.SBNamespace, error) {
```

Comment:
Can you switch these slice pointer return values to simple slices (applies to all functions in this file)? Note that slices are already pointers so you can still return `nil`.

In general, we try to avoid usage of pointers, especially pointers to slices as it leads to increased complexity unless it is necessary. For example, in this case you will have to do nil checks whenever you access these slices as `nil` is no longer equivalent to `[]` and the dereference in the `range` calls can lead to panics. Additionally, slice pointers are most useful when you want to pass by reference, and the use case for that is usually limited to function args.